api/dbadapters/presto.ts

```diff
@@ -70,44 +62,27 @@ export class PrestoDbAdapter implements IDbAdapter {
                 // The presto client cancels executions if this returns true.
                 return isCancelled;
               },
-              state: (error, queryId, stats) => {
-                if (cancelIfError(error)) {
-                  return;
-                }
-                result.queryId = queryId;
-                result.stats = stats;
-              },
-              columns: (error, columns) => {
-                if (cancelIfError(error)) {
-                  return;
-                }
-                result.columns = columns;
-              },
               data: (error, data, columns, stats) => {
-                if (cancelIfError(error)) {
+                if (rejectIfError(error)) {
                   return;
                 }
-                result.columns = columns;
-                result.stats = stats;
                 if (!allRows.push(data)) {
                   isCancelled = true;
                 }
-                result.data = allRows.rows;
-                resolve(result);
+                resolve(allRows.rows);
               },
               success: (error, stats) => {
-                if (cancelIfError(error)) {
+                if (rejectIfError(error)) {
                   return;
                 }
-                result.stats = stats;
-                resolve(result);
+                resolve(allRows.rows);
               },
-              error: cancelIfError
+              error: rejectIfError
             });
           })
       })
       .promise();
-    return { rows: prestoResult.data, metadata: {} };
+    return { rows: rows as any[], metadata: {} };
```

Comment:
this `as any[]` is a code smell, we're just overriding TS's type checking. if you change `new Promise` to `new Promise<any[]>`, this shouldn't be necessary